realm/realm-library/src/androidTest/java/io/realm/RealmResultsTests.java

```diff
@@ -539,6 +539,7 @@ public void onChange(RealmResults<AnnotationIndexTypes> object) {
     }
 
     @Test
+    @RunTestInLooperThread
```

Comment:
the exception thrown order changed. without calling it in the looper thread, it will throw the `wrong thread` first since it is calling `distinctAsync`